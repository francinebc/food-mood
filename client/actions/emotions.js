import request from 'superagent'

const url = 'http://localhost:3000/api/v1/emotions'

// had to have api in here because javascript does not know what a function is
export function getEmotions () {
  return function (dispatch) {
    dispatch(requestEmotions())
    request
      .get(`${url}/`)
      .then(res => res.body)
      .then(res => {
        dispatch(receiveEmotions(res))
      })
      .catch(err => {
        console.log(err)
      })
  }
}

function requestEmotions () {
  return {
    type: 'REQUESTING_EMOTIONS'
  }
}

function receiveEmotions (emotions) {
  return {
    type: 'RECEIVING_EMOTIONS',
    emotions
  }
}
