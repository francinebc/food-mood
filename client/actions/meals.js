// export const ADD_RECIPE = 'ADD_RECIPE'
// export const REMOVE_MEAL = 'REMOVE_MEAL'

// export const addMeal = ({day, meal}) => {
//   return {
//     type: ADD_RECIPE,
//     day,
//     meal
//   }
// }

// export const removeMeal = ({day, meal}) => {
//   return {
//     type: REMOVE_MEAL,
//     day,
//     meal
//   }
// }
