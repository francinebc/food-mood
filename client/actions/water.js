export const updateWater = (water) => {
  return {
    type: 'UPDATE_WATER',
    water
  }
}

export function clearWater () {
  return {
    type: 'CLEAR_WATER'
  }
}