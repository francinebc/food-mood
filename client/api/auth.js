import request from 'superagent'

const url = 'http://localhost:3000/api/v1/auth'

export function register (user) {
  return request
    .post(`${url}/register`)
    .send(user)
    .then(res => res.body)
}

export const signin = (user) => {
  return request
    .post(`${url}/signin`)
    .send({user})
    .then(res => res.body)
}
