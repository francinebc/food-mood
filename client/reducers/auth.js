import {getToken, clearToken} from '../utils/tokens'

const initialState = {
  error: null,
  loggedIn: !!getToken(),
  newRegistration: false,
  pending: false,
  userId: -1
}

export default function authReducer (state = initialState, action) {
  switch (action.type) {
    case 'SIGNIN_PENDING':
    case 'REGISTER_PENDING':
      return {
        ...state,
        loggedIn: false,
        pending: true
      }
    case 'SIGNIN_SUCCESS':
      return {
        ...state,
        error: null,
        loggedIn: true,
        pending: false,
        userId: action.id
      }
    case 'REGISTER_SUCCESS':
      console.log(action)
      return {
        ...state,
        error: null,
        newRegistration: true,
        loggedIn: true,
        pending: false,
        userId: action.id
      }
    case 'SIGNIN_ERROR':
    case 'REGISTER_ERROR':
      return {
        ...state,
        error: true,
        loggedIn: false,
        pending: false,
        userId: -1
      }
    case 'LOGOUT':
      clearToken()
      return {
        ...state,
        loggedIn: false,
        userId: -1
      }
    default:
      return state
  }
}
