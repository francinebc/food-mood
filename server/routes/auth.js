const express = require('express')
const db = require('../db/users')
const router = express.Router()
const token = require('../auth/token')
const hash = require('../auth/hash')




module.exports = router