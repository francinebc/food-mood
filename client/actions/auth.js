import {setToken} from '../utils/tokens'
import {register as registerApi, signin as signinApi} from '../api/auth'

export const signinPending = () => {
  return {
    type: 'SIGNIN_PENDING'
  }
}

export const signinSuccess = (id) => {
  return {
    type: 'SIGNIN_SUCCESS',
    id
  }
}

export const signinError = error => {
  return {
    type: 'SIGNIN_ERROR',
    error
  }
}

export const registerPending = () => {
  return {
    type: 'REGISTER_PENDING'
  }
}

export const registerSuccess = (id) => {
  return {
    type: 'REGISTER_SUCCESS',
    id
  }
}

export const registerError = error => {
  return {
    type: 'REGISTER_ERROR',
    error
  }
}

export const signin = (user) => dispatch => {
  dispatch(signinPending())
  return signinApi(user)
    .then(res => {
      setToken(res.body.token)
      dispatch(signinSuccess(res.body.id))
    })
    .catch(err => dispatch(signinError(err.response.body.error)))
}

export const register = (user) => dispatch => {
  dispatch(registerPending())
  return registerApi(user)
    .then(res => {
      setToken(res.body.token)
      dispatch(registerSuccess(res.body.id))
    })
    .catch(err => dispatch(registerError(err.response.body.error)))
}

export const logout = _ => {
  return {
    type: 'LOGOUT'
  }
}