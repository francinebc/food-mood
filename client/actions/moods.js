import {addMoodApi, getMostRecentMoodApi} from '../api/moods'

export function getMostRecentMood (userId) {
  return function (dispatch) {
    dispatch(requestMood())
    getMostRecentMoodApi(userId)
      .then(res => {
        const mood = res
        dispatch(receiveMood(mood))
      })
  }
}

// Various actions to do with mood
function requestMood () {
  return {
    type: 'REQUESTING_MOOD'
  }
}

function receiveMood (mood) {
  return {
    type: 'RECEIVING_MOOD',
    mood
  }
}

export function saveNewMood (mood, currentMealId) {
  return function (dispatch) {
    // we're optimistic ;)
    dispatch(requestAddMood())
    addMoodApi(mood, currentMealId)
      .then((id) => {
        mood.id = id
        mood.meal_id = currentMealId
        dispatch(updateMood(mood))
        dispatch(receiveAddMood())
      })
      .catch(err => {
        if (err) dispatch(receiveAddMood())
      })
  }
}

// Actions which request, receibe and save moods
function requestAddMood () {
  return {
    type: 'REQUESTING_ADD_MOOD'
  }
}

function receiveAddMood () {
  return {
    type: 'RECEIVING_ADD_MOOD'
  }
}

export const updateMood = (mood) => {
  return {
    type: 'UPDATE_MOOD',
    mood
  }
}
