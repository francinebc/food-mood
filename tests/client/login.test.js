import nock from 'nock'
import * as auth from '../../client/actions/auth'

let state = null

beforeAll(() => {
  nock(/.*/)
    .persist()
    .post('/api/v1/auth/login', {username: 'abc', password: '123'})
    .reply(201, {ok: true, token: '0123456789ABCDEF'})
    .post('/api/v1/auth/login', {username: 'abc'})
    .reply(401, {ok: false, error: 'No password provided.'})
    .post('/api/v1/auth/login', {username: 'abc', password: '123', email: 'bar@foo'})
    .reply(201, {ok: true, token: '0123456789ABCDEF'})
    .post('/api/v1/auth/login', {username: 'xyz', password: '321'})
    .reply(201, {ok: false, error: 'No email provided'})
})

afterAll(() => {
  nock.cleanAll()
})

beforeEach(() => {
  state = {
    error: null,
    loggedIn: false,
    newRegistration: false,
    pending: false
  }
})

test('signinPending matches the last snapshot', () => {
  expect(auth.signinPending()).toMatchSnapshot()
})

test('signinSuccess matches the last snapshot', () => {
  expect(auth.signinSuccess()).toMatchSnapshot()
})

test.skip('signinError matches the last snapshot', () => {
  expect(auth.signinError('something wrong happend')).toMatchSnapshot()
})

// To fix signin
test.skip('sign in dispatches PENDING', () => {
  expect.assertions(1)
  const expected = {type: 'SIGNIN_SUCCESS'}
  const dispatch = jest.fn()
  return auth.signin('abc', '123')(dispatch)
    .then(() => expect(dispatch).toHaveBeenCalledWith(expected))
})

test('sign in dispatches SUCCESS', () => {
  expect.assertions(1)
  const expected = {type: 'SIGNIN_PENDING'}
  const dispatch = jest.fn()
  return auth.signin('abc', '123')(dispatch)
    .then(() => expect(dispatch).toHaveBeenCalledWith(expected))
})

test.skip('sign in dispatches FAILURE on error', () => {
  expect.assertions(1)
  const expected = {type: 'SIGNIN_ERROR', error: 'No password provided'}
  const dispatch = jest.fn()
  return auth.signin('abc')(dispatch)
    .then(() => expect(dispatch).toHaveBeenCalledWith(expected))
})
