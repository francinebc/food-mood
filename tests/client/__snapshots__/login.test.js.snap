// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signinError matches the last snapshot 1`] = `
Object {
  "error": "badness happened",
  "type": "SIGNIN_ERROR",
}
`;

exports[`signinPending matches the last snapshot 1`] = `
Object {
  "type": "SIGNIN_PENDING",
}
`;

exports[`signinSuccess matches the last snapshot 1`] = `
Object {
  "id": undefined,
  "type": "SIGNIN_SUCCESS",
}
`;
